modules/extensions-support/src/main/java/org/mule/module/extension/internal/config/ExtensionConfig.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.config;
+
+import org.mule.time.Time;
+
+/**
+ * A config object which gathers all the configuration properties about how
+ * the platform should handle extensions
+ *
+ * @since 4.0
+ */
+public interface ExtensionConfig
```

Comment:
The actual instance of extension config is parsed using mule's legacy parsers which require the setter. I don't want to expose the setter so I use the interface